src/App.tsx

```diff
@@ -15,35 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  BY_LENGTH = 'length',
+  ALPHABETICALLY = 'alphabetically',
+  DEFAULT = '',
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.DEFAULT);
+  const [sortOrder, setSortOrder] = useState<boolean>(true);
+
+  function prepareGoods(goods: string[], sortBy: string, order = true) {
+    const preparedGoods = [...goods];
+
+    switch (sortBy) {
+      case SortType.BY_LENGTH: {
+        preparedGoods.sort((goodA, goodB) => goodA.length - goodB.length);
+        break;
+      }
+
+      case SortType.ALPHABETICALLY: {
+        preparedGoods.sort((goodA, goodB) => goodA.localeCompare(goodB));
+        break;
+      }
+
+      default:
+        break;
+    }
+
+    return order ? preparedGoods : preparedGoods.reverse();
+  }
+
+  function resetHandler(): void {
+    setSortField(SortType.DEFAULT);
+    setSortOrder(true);
+  }
+
+  const visibleGoods = prepareGoods(goodsFromServer, sortField, sortOrder);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
-          Sort alphabetically
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortField !== SortType.ALPHABETICALLY,
+          })}
+          onClick={() => setSortField(SortType.ALPHABETICALLY)}
+        >
+          Sort {SortType.ALPHABETICALLY}
         </button>
 
-        <button type="button" className="button is-success is-light">
-          Sort by length
+        <button
+          type="button"
+          className={classNames('button', 'is-success', {
+            'is-light': sortField !== SortType.BY_LENGTH,
+          })}
+          onClick={() => setSortField(SortType.BY_LENGTH)}
+        >
+          Sort by {SortType.BY_LENGTH}
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-warning', {
+            'is-light': sortOrder !== false,
+          })}
+          onClick={() => {
+            setSortOrder(prevOrder => !prevOrder);
+          }}
+        >
           Reverse
         </button>
```

Comment:
The active button for reversing the order should be marked to indicate that it is the active action. Consider using a different style or attribute to denote the active state in accordance with [UX] guidelines.